### PR TITLE
Data type constraints documentation

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Add data type constraints documentation.

### Pull request type

Please check what kind of PR this is:

- [ ] Fix or improvement of an existing page
- [x] New documentation page, release related

### Related PRs and issues

PR this doc page is related to: 
[PR2289](https://github.com/memgraph/memgraph/pull/2289)

Closes:

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

File: pages/fundamentals/constraints.mdx [hunk 1]
```diff
@@ -161,6 +161,102 @@ DROP CONSTRAINT ON (n:Employee) ASSERT n.name, n.address IS UNIQUE;
 
 Now, `SHOW CONSTRAINT INFO;` returns an empty set.
 
+## Data type constraint
+
+The data type constraint enforces that each label-property pair is of a certain data type.
```
Comment (hal-eisen-MG) [inline pages/fundamentals/constraints.mdx:166]: There's a corner case I'm not seeing addressed here. What if two different labels apply different constraints to the same property?

CREATE (n:Circle:Blue {name: "John Doe", age: 30});
CREATE CONSTRAINT ON (n:CIrcle) ASSERT n.age IS TYPED integer;
CREATE CONSTRAINT ON (n:Blue) ASSERT n.age IS TYPED float;

What happens here?

File: pages/fundamentals/constraints.mdx [hunk 2]
```diff
@@ -161,6 +161,102 @@ DROP CONSTRAINT ON (n:Employee) ASSERT n.name, n.address IS UNIQUE;
...
+| data_type       | Person       | name            |   STRING    |
+| data_type       | Person       | surname         |   STRING    |
++-----------------+--------------+-----------------+-------------|
+```
+
+To drop the created constraints, use the following queries:
+
+```cypher
+DROP CONSTRAINT ON (n:Person) ASSERT n.name IS TYPED STRING;
+DROP CONSTRAINT ON (n:Person) ASSERT n.surname IS TYPED STRING;
```
Comment (hal-eisen-MG) [inline pages/fundamentals/constraints.mdx:254]: Don't repeat the STRING datatype in the 2nd example

File: pages/fundamentals/constraints.mdx (hunk 1 above)
Comment (imilinovic) [reply pages/fundamentals/constraints.mdx:166]: Covered below by
`You can only have one data type constraint on a given label-property pair.`

File: pages/fundamentals/constraints.mdx (hunk 2 above)
Comment (imilinovic) [reply pages/fundamentals/constraints.mdx:254]: Changed with n.age and INTEGER

File: pages/fundamentals/constraints.mdx (hunk 1 above)
Comment (hal-eisen-MG) [reply pages/fundamentals/constraints.mdx:166]: Sorry, I'm confused. What happens when I try to store 5.5 into the `age` property? Does it work or does it violate the constraint?

This is not just about "can the constraint exist" - it is also about how does the constraint behave later in time during normal usage.

My mental model is that each node has a single property store. Is that correct?

Or is there an extra property store for each label on the node? So a node with 5 labels would have 5 property stores?

**Review by hal-eisen-MG — APPROVED** [review]
LGTM

**Pull request merged**